pkg/catalogsourceconfig/succeeded.go

```diff
@@ -56,9 +61,23 @@ func (r *succeededReconciler) Reconcile(ctx context.Context, in *v2.CatalogSourc
 		out.Status = v2.CatalogSourceConfigStatus{}
 		nextPhase = phase.GetNext(phase.Configuring)
 		msg = "Child resource(s) have been deleted, scheduling for configuring"
+		return
+	}
+
+	needsProxyUpdate, err := watches.CheckProxyResource(r.client, in.Name, in.Namespace)
+	if err != nil {
+		return
 	}
 
-	r.log.Info(msg)
+	if needsProxyUpdate {
+		// The Environment Variables in the deployment created by the CatalogSourceConfig
+		// are out of sync with those in the Cluster Proxy.
+		// Drop the existing Status field so that reconciliation can start anew.
+		out.Status = v2.CatalogSourceConfigStatus{}
+		nextPhase = phase.GetNext(phase.Configuring)
+		msg = "Proxy environment variables not in sync, scheduling for configuring"
```

Comment:
That is a great idea and will be implemented in the next update to the PR. Perhaps we could add it to the `watches` package so it can be shared between the `CSC` and `opsrc` packages? Otherwise I will make a function in this file for each.

Reply:
This specific to `CatalogSourceConfigs` and `OperatorSources`. I would suggest adding it to `*_types.go` similar to `EnsureGVK()` etc